### PR TITLE
WIP: Test patch for ARROW-15444

### PR DESCRIPTION
**Pull request by xhochy**

### CODE REVIEW

**Comment by xhochy** [comment]
@github-actions crossbow submit conda-linux-gcc-py37-ppc64le

File: cpp/src/arrow/compute/kernels/aggregate_internal.h [hunk 1]
```diff
@@ -173,7 +174,7 @@ enable_if_t<std::is_floating_point<SumType>::value, SumType> SumArray(
       block_sum = sum[cur_level];
       sum[cur_level] = 0;
       ++cur_level;
-      DCHECK_LT(cur_level, levels);
+      assert(cur_level, levels);
```
Comment (pitrou) [inline cpp/src/arrow/compute/kernels/aggregate_internal.h:177]:
```suggestion
      assert(cur_level < levels);
```

**Comment by pitrou** [comment]
@github-actions crossbow submit conda-linux-gcc-py37-ppc64le

**Comment by xhochy** [comment]
@github-actions crossbow submit conda-linux-gcc-py37-ppc64le

**Comment by xhochy** [comment]
@pitrou Can you point me to one of the Ubuntu GCC 7.5 builds? I'd love to compare the logs.

**Comment by xhochy** [comment]
@github-actions crossbow submit conda-linux-gcc-py37-ppc64le

**Comment by pitrou** [comment]
@xhochy Here is one: https://github.com/ursacomputing/crossbow/runs/4940107692?check_suite_focus=true

**Pull request closed without merging**